Added SQLite3 sink to the solution

PedroRod opened the pull request:
You can find more info on how to implement it here [https://github.com/PedroRod/sqlite3_database_sink](url)

gabime commented on include/spdlog/sinks/database_logger_sink.h:
```diff
@@ -0,0 +1,156 @@
+
+struct database_schema
+{
+	std::string TableName;
```
To be consistent with spdlog's naming convention, please use snake_case for all variable names ("table_name", "column_map", "column_name", "value", "thread_id" ,etc..)

gabime commented on include/spdlog/sinks/database_logger_sink.h:
```diff
@@ -0,0 +1,156 @@
+				}
+
+				auto query = fmt::format("INSERT INTO {0} ({1},{3},{5},{7},{9}) VALUES ('{2}','{4}','{6}','{8}',{10})",
+					_schema.TableName,
```
This is sql injection waiting to happen.. Please use sql params instead

gabime commented:
Thanks. I added my comments in places that need to be fixed

gabime commented on include/spdlog/sinks/database_logger_sink.h:
```diff
@@ -0,0 +1,156 @@
+
+			}
+
+			explicit database_logger_sink(const std::string& databaseName)
```
Please place ctors first, before all other public metthods

PedroRod commented:
I made some adjustments based on your comments.

You are totally right about the sql injection, I did not take that into consideration since for my application I do not need to worry about it, but you are totally right.
That got changed, changed code style to use same one you use for spdlog and removed the column mapping since there is no clean way of mapping columns in a prepared statement.

gabime commented:
Thanks. prepare_statement  should be called only once (e.g. upon ctor and the resultant sqlite3_stmt* stored as private member to be binded upon each call). Otherwise the performance will suffer. 
Also, please rename the sink and .h to sqlite_sink please
